src/Microsoft.TemplateEngine.Tasks/Utilities/MSBuildLogger.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.TemplateEngine.Tasks.Utilities
+{
+    /// <summary>
+    /// Implements an ILogger that passes the logs to the wrapped TaskLoggingHelper.
+    /// </summary>
+    internal sealed class MSBuildLogger : ILogger
+    {
+        private static readonly IDisposable Scope = new DummyDisposable();
+
+        private readonly string _categoryHeader;
+        private readonly TaskLoggingHelper _loggingHelper;
+
+        public MSBuildLogger(string category, TaskLoggingHelper loggingHelperToWrap)
+        {
+            _categoryHeader = category + ": ";
+            _loggingHelper = loggingHelperToWrap;
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => Scope;
+
+        public bool IsEnabled(LogLevel logLevel) => true;
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.Trace:
+                case LogLevel.Debug:
+                case LogLevel.Information:
+                    _loggingHelper.LogMessage(_categoryHeader + formatter(state, exception));
+                    break;
+                case LogLevel.Warning:
+                    _loggingHelper.LogWarning(_categoryHeader + formatter(state, exception));
+                    break;
+                case LogLevel.Error:
+                case LogLevel.Critical:
+                    _loggingHelper.LogError(_categoryHeader + formatter(state, exception));
```

Comment:
should we do better logging here using other [overload ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.tasklogginghelper.logerror?view=msbuild-16-netcore#Microsoft_Build_Utilities_TaskLoggingHelper_LogError_System_String_System_String_System_String_System_String_System_Int32_System_Int32_System_Int32_System_Int32_System_String_System_Object___)? how this message will appear inside VS?

Reply:
We don't have any more information than we provide here. If we use that overload, what other value can we pass to the overload as an argument?

Reply:
I think we can come up with some codes and other fields to get it nicer. 
We can do counter on error codes, and perhaps mentioning the file/line for error. See how it is done in xlf: https://github.com/dotnet/xliff-tasks/blob/0da469fe1c2714d0e25cad1143632562fb528425/src/XliffTasks/Tasks/TaskLoggingHelperExtensions.cs

Reply:
If I remember correctly most of the errors will be related to file/line (e.g. missing ID for post action, etc).